internal/weldr/api_test.go

```diff
@@ -26,16 +29,48 @@ var packages = rpmmd.PackageList{
 	{Name: "package2"},
 }
 
-func testRoute(t *testing.T, api *weldr.API, method, path, body string, expectedStatus int, expectedJSON string) {
+func externalRequest(method, path, body string) *http.Response {
+	client := http.Client{
+		Transport: &http.Transport{
+			DialContext: func(_ context.Context, _, _ string) (net.Conn, error) {
+				return net.Dial("unix", "/run/weldr/api.socket")
+			},
+		},
+	}
+
+	req, _ := http.NewRequest(method, "http://localhost"+path, bytes.NewReader([]byte(body)))
+
+	if method == "POST" {
+		req.Header.Set("Content-Type", "application/json")
+	}
+	resp, _ := client.Do(req)
+
+	return resp
+}
+
+func internalRequest(api *weldr.API, method, path, body string) *http.Response {
 	req := httptest.NewRequest(method, path, bytes.NewReader([]byte(body)))
+
 	if method == "POST" {
 		req.Header.Set("Content-Type", "application/json")
 	}
 	resp := httptest.NewRecorder()
 	api.ServeHTTP(resp, req)
 
-	if resp.Code != expectedStatus {
-		t.Errorf("%s: expected status %v, but got %v", path, expectedStatus, resp.Code)
+	return resp.Result()
+}
+
+func testRoute(t *testing.T, api *weldr.API, method, path, body string, expectedStatus int, expectedJSON string) {
+	var resp *http.Response
+
+	if os.Getenv("EXTERNAL") == "true" {
```

Comment:
Hm sorry to be nitpicky, but could you please namespace the environment variable? Something like `OSBUILD_COMPOSER_TEST_EXTERNAL`. Also, don't check for `true` specifically, but only if it is "set", i.e., a non-empty string.

Reply:
Oh no thanks for the nitpicks! This has been fixed.

Reply:
Thanks! You were checking for existence now, but unsetting a variable like this
```
$ OSBUILD_COMPOSER_TEST_EXTERNAL= go test ./...
```
is quite common. I've changed it to `len(os.Getenv("OSBUILD_COMPOSER_TEST_EXTERNAL")) > 0`.